frontend/lib/api/ApiClient.ts

```diff
@@ -12,27 +12,44 @@ export interface ApiResponse<T> {
   data: T;
 }
 
-export type ApiError = {
+export class ApiError extends Error {
   status: ApiResponseStatus.ClientError | ApiResponseStatus.ServerError;
   code: number;
   error?: string;
-};
+
+  constructor(status: ApiResponseStatus.ClientError | ApiResponseStatus.ServerError, code: number, error?: string) {
+    super(error);
+    this.status = status;
+    this.code = code;
+    this.error = error;
+  }
+
+  withContext(message: string): this {
+    this.error = message;
```

Comment:
maybe set `this.message` as well to not let them diverge? Ideally we would use message only and not have a shadow `error` field..

Reply:
Check -I've merged error and message field.